File: frontend/src/models/viewmodels/friendship.vm.ts
import { FriendshipStatus, PaginationData } from "../app.interface";

export interface FriendshipVM {
  id: number;
  otherUserId: number;
  otherUserName: string;
  otherUserAvatar: string;
  isRequester: boolean;
  status: FriendshipStatus;
}

export interface PagedFriendshipsVM {
  friends: FriendshipVM[];
  pagination: PaginationData;
}

File: backend/Common/ViewModels/FriendshipVMs/FriendshipVM.cs
using Models.Friendships;

namespace Common.ViewModels.FriendshipVMs;
public class FriendshipVM
{
    public int Id { get; set; }
    public int OtherUserId { get; set; }
    public string OtherUserName { get; set; }
    public string OtherUserAvatar { get; set; }
    // Indicates whether the current user sent the request
    public bool IsRequester { get; set; }
    public FriendshipStatus Status { get; set; }
}
